### PR TITLE
Remove MPI requirement from logging/replay

### PR DESCRIPTION
**Pull request by davidbeckingsale**
MPI is now initialized by the user calling `umpire::initialize(MPI_Comm comm)` - once it's initialized, MPI information is added to the logging and replay output.

MPI rank is no longer included in filenames.

MPI must be initialized _before_ using any other Umpire resources if using a custom directory for log/replay output.

### CODE REVIEW

**Comment by jdahm** [comment]
Looks good!

File: src/umpire/util/IOManager.cpp [hunk 1]
```diff
@@ -177,7 +184,7 @@ makeUniqueFilename(
     ss.str("");
     ss.clear();
     unique_id++;
-    ss << base_dir << "/" << name << "." << rank << "." << pid << "." << unique_id << "." << extension;
+    ss << base_dir << "/" << name << "." << pid << "." << unique_id << "." << extension;
```
[the remark this answers is not in the document]
Comment (davidbeckingsale) [reply src/umpire/util/IOManager.cpp]: Possibly. Does it need to be in the filename?

File: src/umpire/util/IOManager.cpp [hunk 2]
```diff
@@ -97,44 +96,51 @@ IOManager::initialize(
...
+          else if ( !(S_ISDIR(info.st_mode)) )
           {
-            UMPIRE_ERROR("mkdir(" << s_root_io_dir << ") failed");
+            UMPIRE_ERROR(s_root_io_dir << "exists and is not a directory");
           }
+#endif
         }
+        MPI::sync();
+      }  else {
+        UMPIRE_ERROR("Cannot create output directory before MPI has been initialized. Please unset UMPIRE_IO_DIR in your environment");
```
[the remark this answers is not in the document]
Comment (davidbeckingsale) [reply src/umpire/util/IOManager.cpp]: Agree.

File: CHANGELOG.md [hunk 3]
```diff
@@ -33,6 +33,8 @@ and a dynamic pool for those that are larger.
 
 - Added a new IOManager that stores logging and replay output to files.
 
+- Added MPI-awareness to output for both logging and replay.
+
```
[the remark this answers is not in the document]
Comment (davidbeckingsale) [reply CHANGELOG.md]: I will update the recipe for coalesce. I don't think `Interface Change` is good, but it can go under `Changed`.

**Comment by davidbeckingsale** [comment]
Also, nothing is lost provided that the first thing you do is access the ResourceManager (which initializes everything).

I would like to address the buffering issue another time - please create a JIRA issue with this concern.

File: src/umpire/util/IOManager.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (mcfadden8) [reply src/umpire/util/IOManager.cpp]: I don't think so.  But I am not sure that it is needed now that I understand that the rank information may be provided later on when/if MPI is initialized.

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (davidbeckingsale) [reply CHANGELOG.md]: Doing this in a different issue.

File: src/umpire/util/IOManager.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidbeckingsale) [reply src/umpire/util/IOManager.cpp]: I have addressed this in the latest commit.

**Pull request merged**